webapp/channels/src/components/admin_console/system_properties/user_properties_dot_menu.tsx

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {CheckIcon, ChevronRightIcon, DotsHorizontalIcon, EyeOutlineIcon, TrashCanOutlineIcon} from '@mattermost/compass-icons/components';
+import type {FieldVisibility, UserPropertyField} from '@mattermost/types/properties';
+
+import * as Menu from 'components/menu';
+
+import './user_properties_dot_menu.scss';
+import {useUserPropertyFieldDelete} from './user_properties_delete_modal';
+import {isCreatePending} from './user_properties_utils';
+
+type Props = {
+    field: UserPropertyField;
+    updateField: (field: UserPropertyField) => void;
+    deleteField: (id: string) => void;
+}
+
+const menuId = 'user-property-field_dotmenu';
+
+const DotMenu = ({
+    field,
+    updateField,
+    deleteField,
+}: Props) => {
+    const {promptDelete} = useUserPropertyFieldDelete();
+
+    const handleDelete = () => {
+        if (isCreatePending(field)) {
+            // skip prompt when field is pending creation
+            deleteField(field.id);
+        } else {
+            promptDelete(field).then(() => deleteField(field.id));
+        }
+    };
+
+    const handleVisibilityChange = (visibility: FieldVisibility) => {
+        updateField({...field, attrs: {...field.attrs, visibility}});
+    };
+
+    let selectedVisibilityLabel;
+
+    if (field.attrs.visibility === 'always') {
```

Comment:
Should we have constants for the visibility attribute values?

Reply:
It's a good idea to consider, though it's backed by TS string literal unions which provide type checking and autocomplete. It's a pattern used elsewhere in webapp, though certainly not everywhere and it does have its pros and cons.